pr_consistency/1.get_merged_prs.py

```diff
@@ -1,116 +1,105 @@
 # The purpose of this script is to download information about all pull
 # requests merged into the master branch of the given repository. This
-# information is downloaded to a JSON file. This includes the 'last
-# modified' date for all pull requests, so that when the script is re-run,
-# only modified or new entries are downloaded. At this time, this script
-# requires the developer version of the pygithub package.
+# information is downloaded to a JSON file.
 
 import os
 import sys
 import json
-
-from datetime import datetime, timedelta
-from github import Github
-
-from common import get_credentials
-
-
-def parse_isoformat(string):
-    return datetime.strptime(string, "%Y-%m-%dT%H:%M:%S")
-
+import requests
+
+QUERY_TEMPLATE = """
+{{
+  repository(owner: "{owner}", name: "{repository}") {{
+    pullRequests(first:100, orderBy: {{direction: ASC, field: CREATED_AT}}, baseRefName: "master", states: MERGED{after}) {{
+      edges {{
+        node {{
+          title
+          number
+          mergeCommit {{
+            oid
+          }}
+          createdAt
+          updatedAt
+          mergedAt
+          milestone {{
+            title
+          }}
+          labels(first: 10) {{
+            edges {{
+              node {{
+                name
+              }}
+            }}
+          }}
+        }}
+        cursor
+      }}
+    }}
+  }}
+}}
+"""
 
 if sys.argv[1:]:
     REPOSITORY = sys.argv[1]
 else:
     REPOSITORY = 'astropy/astropy'
 
+OWNER = os.path.dirname(REPOSITORY)
 NAME = os.path.basename(REPOSITORY)
 
 print("The repository this script currently works with is '{}'.\n"
       .format(REPOSITORY))
 
+json_filename = 'merged_pull_requests_{}.json'.format(NAME)
 
-# Get handle to repository
-g = Github(*get_credentials())
-repo = g.get_repo(REPOSITORY)
+TOKEN = input('Enter your GitHub personal token: ')
```

Comment:
does this require the token to be entered each time, or will reuse it like the previous version did?

Reply:
it requires it every time... which now that you mention it is perhaps a reason not to make this change just yet!

Reply:
Unlike the previous method this requires a token, no way to do it with username/password. We could add ways of specifying it permanently though, env var, hidden file, etc.

Reply:
The previous version also used a token, at least as I recall I used it with a token to work around 2fa (can check on it tomorrow evening)